bugsnag-android-core/src/main/java/com/bugsnag/android/Client.java

```diff
@@ -145,7 +145,10 @@ public Unit invoke(Boolean connected) {
                 logger);
         systemBroadcastReceiver = new SystemBroadcastReceiver(this, logger);
 
-        metadataState = configuration.metadataState.copy(configuration.metadataState.getMetadata());
+        Metadata orig = configuration.metadataState.getMetadata();
+        Metadata copy = orig.copy(orig.toMap(), orig.getJsonStreamer(), orig.getRedactedKeys());
```

Comment:
Should this be the parameterless override of `copy` from `Metadata`?

```suggestion
        Metadata copy = orig.copy();
```

Also, can you either comment on the purpose of this or extract to private function to document.

Reply:
I've updated this to use the parameterless override, and added an inline comment for documentation.